### DISCUSSION TITLE
add_point_labels: center label at point

### DISCUSSION THREAD

**Discussion by nschloe** [pyvista/pyvista#2544] [Q&A] [answered]
I would like to highlight the points in my mesh. To this end, I'm using [`add_point_labels()`](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.add_point_labels.html).
```python
p.add_point_labels(
    coords,
    idx,
    point_size=10,
    shape_color="#555555",
    font_size=12,
    text_color="white",
)
```
![screenshot](https://user-images.githubusercontent.com/181628/166232343-64697f2f-7170-46c7-a86b-a18860379647.png)

Note how the point markers almost disappear beneath the labels. A better representation would be to _not_ mark the points themselves, but to center the labels at the respective coordinates. I can do the former with `show_points=False`.

![screenshot](https://user-images.githubusercontent.com/181628/166232005-f3dea304-f139-4b1c-8b2e-adf8462a86f2.png)

How can I center the labels?

**Comment by MatthewFlamm** [comment, marked as answer]
I do not know how to do this natively in pyvista.  Here is what I tried to get what you want, but it doesn't work perfectly.

The current method has this section:

https://github.com/pyvista/pyvista/blob/f1f4df2a671e22dff233f4c0e2b00aa635d42644/pyvista/plotting/plotting.py#L3850-L3857

If the following lines are added:

```py
textprop.SetJustificationToCentered()
textprop.SetVerticalJustificationToCentered()
```

Then I get something closer to what you are asking, but the box behind isn't aligned correctly.

![image](https://user-images.githubusercontent.com/39341281/166303980-48b5f486-8c5c-4263-8713-9c406051919f.png)

It could be a bug in how I implemented it, or in vtk itself.

**Reply by MatthewFlamm** [reply]
Below is a hacky solution that only works if you don't zoom or rotate and you are willing to spend time tweaking the size of the background box in conjunction with size of labels, zoom, etc.  Quite a lot of caveats, but it does work...

Unfortunately this also requires changing the method `add_point_labels`, and I'm hesitant to add these parameters without figuring out how to fix the background shape.

```py
import pyvista
circ = pyvista.Circle(resolution=12)
plotter = pyvista.Plotter()
plotter.add_mesh(circ, style='wireframe')
plotter.add_point_labels(circ, labels=list(range(circ.n_points)), font_size=16, shape=None)
plotter.add_mesh(circ.glyph(geom=pyvista.Plane(i_size=0.045, j_size=0.04, i_resolution=1, j_resolution=1)))
plotter.view_xy()
plotter.show()
```



![image](https://user-images.githubusercontent.com/39341281/166325049-2a52d2d2-c14f-4c57-815f-65f6fd717a8f.png)
